src/common/snippets/src/pass/matmul_to_brgemm.cpp

```diff
@@ -17,41 +17,46 @@ namespace ov {
-        const auto& tensor = utils::pshape_to_vdims(input.get_partial_shape());
-        const auto& subtensor = get_subtensor();
-        lowered::PortDescriptorUtils::set_port_descriptor_ptr(input, std::make_shared<lowered::PortDescriptor>(tensor, subtensor));
-    }
-    const auto& tensor = utils::pshape_to_vdims(brgemm->get_output_partial_shape(0));
-    const auto& subtensor = get_subtensor();
-    lowered::PortDescriptorUtils::set_port_descriptor_ptr(brgemm->output(0), std::make_shared<lowered::PortDescriptor>(tensor, subtensor));
-}
+using namespace lowered;
 
 MatMulToBrgemm::MatMulToBrgemm() {
     MATCHER_SCOPE(MatMulToBrgemm);
     auto matmul_pattern = ov::pass::pattern::wrap_type<ov::opset1::MatMul>({ov::pass::pattern::any_input(), ov::pass::pattern::any_input()});
 
-    auto callback = [OV_CAPTURE_CPY_AND_THIS](ov::pass::pattern::Matcher& m) {
+    auto callback = [matmul_pattern](ov::pass::pattern::Matcher& m) {
         OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "ov::intel_cpu::pass::MatMulToBrgemm")
-        auto& pm = m.get_pattern_value_map();
+        const auto& pm = m.get_pattern_value_map();
         const auto matmul = as_type_ptr<ov::opset1::MatMul>(pm.at(matmul_pattern).get_node_shared_ptr());
         // Brgemm doesn't support transposed inputs currently, so we don't convert such matmuls
-        if (matmul->get_transpose_a() || matmul->get_transpose_b())
+        if (matmul->get_transpose_a())
             return false;
 
-        auto brgemm = std::make_shared<op::Brgemm>(matmul->get_input_source_output(0), matmul->get_input_source_output(1));
+        auto generate_layout = [](const ov::PartialShape& shape, const bool transpose) {
+            std::vector<size_t> layout(shape.size());
+            std::iota(layout.begin(), layout.end(), 0);
+            if (transpose)
+                std::swap(*layout.rbegin(), *(layout.rbegin() + 1));
+            return layout;
+        };
+
+        const auto& pshape_a = matmul->get_input_partial_shape(0);
+        const auto& pshape_b = matmul->get_input_partial_shape(1);
+        const auto layout_a = generate_layout(pshape_a, matmul->get_transpose_a());
+        const auto layout_b = generate_layout(pshape_b, matmul->get_transpose_b());
+        const auto brgemm = std::make_shared<op::Brgemm>(matmul->input_value(0), matmul->input_value(1), 0, 0, 0, layout_a, layout_b);
+
+        static const std::vector<size_t> subtensor{PortDescriptor::ServiceDimensions::FULL_DIM, PortDescriptor::ServiceDimensions::FULL_DIM};
+        PortDescriptorUtils::set_port_descriptor_ptr(brgemm->input(0), std::make_shared<PortDescriptor>(pshape_a, subtensor, layout_a));
+        PortDescriptorUtils::set_port_descriptor_ptr(brgemm->input(1), std::make_shared<PortDescriptor>(pshape_b, subtensor, layout_b));
+        PortDescriptorUtils::set_port_descriptor_ptr(brgemm->output(0), std::make_shared<PortDescriptor>(brgemm->get_output_partial_shape(0), subtensor));
+
```

Comment:
> We need to execute all this code because of a particular implementation of port descriptors in snippets. Ideally, the transformation should not care about these low-level details, the node should be responsible for default initialization of its port descriptors. This way we can decouple transformation pipeline from the implementation details of particular features.
As far as I remember, we can't do it directly in the node's constructor (because outputs are not yet created)

Exactly. Regarding a dedicated static function creation: I think we can't reuse it in derived nodes because here we set 
 specific layouts based on `get_transpose_a/b`: this logic can be used only here. In the derived nodes, we copy port descriptors from Brgemm node, or set new port descriptors with FULL_DIM subtensors. So it's similar but not the same logic, and that's not clear for me how we can nicely reuse some static function.